src/test/java/de/tum/in/www1/artemis/legal/DataExportCreationServiceTest.java

```diff
@@ -0,0 +1,466 @@
+import de.tum.in.www1.artemis.user.UserUtilService;
+import de.tum.in.www1.artemis.util.FileUtils;
+import de.tum.in.www1.artemis.util.ZipFileTestUtilService;
+
+class DataExportCreationServiceTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    private static final String TEST_PREFIX = "dataexportcreation";
+
+    private static final String FILE_FORMAT_TXT = ".txt";
+
+    private static final String FILE_FORMAT_PDF = ".pdf";
+
+    private static final String FILE_FORMAT_ZIP = ".zip";
+
+    private static final String FILE_FORMAT_CSV = ".csv";
+
+    @Value("${artemis.repo-download-clone-path}")
+    private Path repoDownloadClonePath;
+
+    @Autowired
+    private ZipFileTestUtilService zipFileTestUtilService;
+
+    @Autowired
+    private ProgrammingExerciseTestService programmingExerciseTestService;
+
+    @Autowired
+    private ExerciseUtilService exerciseUtilService;
+
+    @Autowired
+    private ExamUtilService examUtilService;
+
+    @Autowired
+    private CourseUtilService courseUtilService;
+
+    @Autowired
+    private ExamRepository examRepository;
+
+    @Autowired
+    private StudentExamRepository studentExamRepository;
+
+    @Autowired
+    private ExerciseRepository exerciseRepository;
+
+    @Autowired
+    private DataExportRepository dataExportRepository;
+
+    @Autowired
+    private DataExportCreationService dataExportCreationService;
+
+    @Autowired
+    private ApollonRequestMockProvider apollonRequestMockProvider;
+
+    @Autowired
+    @Qualifier("apollonRestTemplate")
+    private RestTemplate restTemplate;
+
+    @Autowired
+    private ApollonConversionService apollonConversionService;
+
+    @Autowired
+    private UserUtilService userUtilService;
+
+    @Autowired
+    private ProgrammingExerciseUtilService programmingExerciseUtilService;
+
+    @Autowired
+    private ParticipationUtilService participationUtilService;
+
+    @Autowired
+    private ConversationUtilService conversationUtilService;
+
+    @Autowired
+    private QuizExerciseUtilService quizExerciseUtilService;
+
+    @Autowired
+    private PostRepository postRepository;
+
+    @Autowired
+    private AnswerPostRepository answerPostRepository;
+
+    @BeforeEach
+    void initTestCase() throws IOException {
+        userUtilService.addUsers(TEST_PREFIX, 2, 5, 0, 1);
+        userUtilService.adjustUserGroupsToCustomGroups(TEST_PREFIX, "", 2, 5, 0, 1);
+
+        apollonConversionService.setRestTemplate(restTemplate);
+
+        apollonRequestMockProvider.enableMockingOfRequests();
+
+        // mock apollon conversion 8 times, because the last test includes 8 modeling exercises
+        mockApollonConversion();
+        mockApollonConversion();
+        mockApollonConversion();
+        mockApollonConversion();
+        mockApollonConversion();
+        mockApollonConversion();
+        mockApollonConversion();
+        mockApollonConversion();
```

Comment:
Can you move this to only the one test that needs the mocking? Also a for loop would be cleaner.

Reply:
all the tests need the mocking, they just differ in the amount of mocked requests needed but moved it to a loop.